### PR TITLE
Add `getStateBuilders` beacon API endpoint 

### PR DESCRIPTION
**Pull request by satushh**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

- Implements the new `getStateBuilders` endpoint from [ethereum/beacon-APIs#614](https://github.com/ethereum/beacon-APIs/pull/614/changes): returns the Gloas builder registry filterable by ids (pubkeys or indices) and statuses (pending/active/exited), 400 for pre-Gloas states. 
- Also fixes builder `version` JSON serialization to the spec's decimal `Uint8` (was hex).
- Tested with unit tests and a live kurtosis gloas devnet where assertoor's builder-lifecycle playbook drove real builders.

**Which issue(s) does this PR fix?**

Fixes #

**Other notes for review**

```
participants:
    - el_type: geth
      el_image: ethpandaops/geth:master
      cl_type: prysm
      cl_image: prysm-bn-custom-image:minimal-build
      vc_image: prysm-vc-custom-image:minimal-build
      validator_count: 128
    - el_type: geth
      el_image: ethpandaops/geth:master
      cl_type: prysm
      cl_image: prysm-bn-custom-image:minimal-build
      vc_image: prysm-vc-custom-image:minimal-build
      validator_count: 128

  network_params:
    slot_duration_ms: 12000
    gloas_fork_epoch: 2
    preset: minimal
    withdrawal_type: "0x01"
    validator_balance: 40000
    num_validator_keys_per_node: 128

  additional_services:
    - dora
    - assertoor

  assertoor_params:
    tests:
      - { file: "https://raw.githubusercontent.com/ethpandaops/assertoor/refs/heads/master/playbooks/gloas-dev/builder-lifecycle.yaml" }

  snooper_enabled: false
  global_log_level: debug

  port_publisher:
    additional_services:
      enabled: true

  dora_params:
    image: ethpandaops/dora:master
```

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have included a uniquely named [changelog fragment file](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md#maintaining-changelogmd).
- [x] I have added a description with sufficient context for reviewers to understand this PR.
- [x] I have tested that my changes work as expected and I added a testing plan to the PR description (if applicable).

### CODE REVIEW

File: beacon-chain/rpc/endpoints.go [hunk 1]
```diff
@@ -963,6 +963,17 @@ func (s *Service) beaconEndpoints(
 			handler: server.GetProposerLookahead,
 			methods: []string{http.MethodGet},
 		},
+		{
+			template: "/eth/v1/beacon/states/{state_id}/builders",
+			name:     namespace + ".GetStateBuilders",
+			middleware: []middleware.Middleware{
+				middleware.ContentTypeHandler([]string{api.JsonMediaType}),
```
Comment (james-prysm) [inline beacon-chain/rpc/endpoints.go:970]: I think you are not handling the empty body case, can we have a test for that and have that added?

File: api/server/structs/conversions_gloas.go [hunk 2]
```diff
@@ -55,9 +55,14 @@ func BuildersFromConsensus(builders []*ethpb.Builder) []*Builder {
 }
 
 func BuilderFromConsensus(b *ethpb.Builder) *Builder {
+	// The builder version is a Uint8 in the beacon API spec, serialized as a decimal string.
+	var builderVersion uint8
+	if len(b.Version) > 0 {
```
Comment (james-prysm) [inline api/server/structs/conversions_gloas.go]: can this actually happen in production?

File: beacon-chain/rpc/endpoints.go (hunk 1 above)
Comment (syjn99) [reply beacon-chain/rpc/endpoints.go:970]: Yeah I think this is true, even if an user sends a POST request with empty body, he/she should explicitly set `Content-Type` header, otherwise Prysm will return `415 Unsupported Media Type`. Maybe we should change `ContentTypeHandler` implementation to handle this (tl;dr - makes `curl -X POST localhost:3500/eth/v1/beacon/states/head/builders` succeed)?

cc. @james-prysm

Comment (satushh) [reply beacon-chain/rpc/endpoints.go:970]: [Fixed](https://github.com/OffchainLabs/prysm/pull/17491/changes/639732347f1527fe7945e400e7549d934d953430) via Jun's suggestion: ContentTypeHandler now skips the check when the body is empty, so curl -X POST .../builders works. Also added middleware tests for the empty and non-empty body cases.

File: api/server/structs/conversions_gloas.go (hunk 2 above)
Comment (satushh) [reply api/server/structs/conversions_gloas.go]: It can't happen in production currently. If you want I can remove the check?

Comment (james-prysm) [reply api/server/structs/conversions_gloas.go]: might as well remove it if it can't happen

**Review by prestonvanloon — DISMISSED** [review]

**Review by james-prysm — APPROVED** [review]

**Review by syjn99 — APPROVED** [review]
